### PR TITLE
Renaming resources as per code changes

### PR DESCRIPTION
**Pull request by Vasundharashukla**
1. **What this PR does / why we need it**:

The Documents and examples needs to be updated with the code enhancements, more example cases have been added. 
Majorly, the documents should require to reflect the renamed resource variables.

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**

4. **Special notes for your reviewer**

### CODE REVIEW

File: templates/data-sources/cluster_node_pool.md.tmpl [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-Create, delete, and update nodepools for your clusters using this Terraform module.
+Create, delete, and update node pools for your clusters using this Terraform module.
 
 For clusters that you create in Tanzu Mission Control, you can define a pool of worker nodes on which your workloads can run
 because Tanzu Mission Control cannot provision additional resources in a cluster that is created elsewhere and subsequently attached, you cannot create a node pool in an attached cluster.
 
 ## Example Usage
 
-{{ tffile "examples/resources/tmc_nodepool/resource.tf" }}
+{{ tffile "examples/data-sources/cluster_node_pool/data-source.tf" }}
```
Comment (shobha2626) [inline templates/data-sources/cluster_node_pool.md.tmpl:16]: resource template is renamed to data-source template?

**Comment by shobha2626** [comment]
Please rebase

File: templates/data-sources/cluster_node_pool.md.tmpl (hunk 1 above)
Comment (Vasundharashukla) [reply templates/data-sources/cluster_node_pool.md.tmpl:16]: It's the data-source template, should have the data-source example, right?

File: docs/data-sources/cluster.md [hunk 2]
```diff
@@ -220,7 +220,7 @@ Optional:
 
 - **cloud_label** (Map of String) Cloud labels
 - **node_label** (Map of String) Node labels
-- **tkg_service_vsphere** (Block List, Max: 1) Nodepool config for Tanzu Kubernetes Grid service vsphere (see [below for nested schema](#nestedblock--spec--tkg_service_vsphere--topology--node_pools--spec--tkg_service_vsphere))
+- **tkg_service_vsphere** (Block List, Max: 1) Nodepool config for tkg service vsphere (see [below for nested schema](#nestedblock--spec--tkg_service_vsphere--topology--node_pools--spec--tkg_service_vsphere))
```
Comment (shreyassreenivas) [inline docs/data-sources/cluster.md:223]: here the tkg is shortform . why was this change made

File: templates/data-sources/cluster_node_pool.md.tmpl (hunk 1 above)
Comment (shobha2626) [reply templates/data-sources/cluster_node_pool.md.tmpl:16]: data-source is renamed to resource. So resolving the comment

**Review by shobha2626 — DISMISSED** [review]
LGTM

**Review by shreyassreenivas — DISMISSED** [review]

**Review by shreyassreenivas — APPROVED** [review]

**Review by shobha2626 — APPROVED** [review]

**Pull request merged**